sim/gui.py

```diff
@@ -34,19 +34,132 @@ def __add__(self, other: object) -> Self:
         return type(self)(self.row + other.row, self.col + other.col)
 
 
-class GUIRenderer:
-    """Class for rendering a maze on screen"""
-    wall_thickness: int = 5
-    wall_color: str = 'red'
-    robot_main_color: str = 'blue'
-    robot_second_color: str = 'yellow'
-    goal_color: str = 'green'
-    heatmap_colors: list = ['blue', 'cyan', 'green', 'yellow', 'orange', 'red', 'brown']
+class GUIRenderer:  # pylint: disable=too-many-instance-attributes
+    """Class for rendering the GUI on screen"""
+
+    def __init__(self, sim: Simulator):
+        """Initialize GUI
+
+        Args:
+            sim (Simulator): The simulation to show.
+        """
+        pg.init()
+        initial_screen_size = (1280, 720)
+        screen = pg.display.set_mode(initial_screen_size, pg.RESIZABLE)
+        pg.display.set_caption('Micromouse')
+        self.screen = screen
+        self.wall_thickness: int = 5
+        self.wall_color: str = 'red'
+        self.robot_main_color: str = 'blue'
+        self.robot_second_color: str = 'yellow'
+        self.goal_color: str = 'green'
+        self.heatmap_colors: list = ['blue', 'cyan', 'green', 'yellow', 'orange', 'red', 'brown']
+        self.ui_manager = pygame_gui.UIManager((screen.get_width(), screen.get_height()))
+        self.start_button = pygame_gui.elements.UIButton(relative_rect=pg.Rect((0, 0), (0, 0)), text='Start', manager=self.ui_manager)
+        self.start_button.tool_tip_text = "Shortcut: 's'"
+        self.step_button = pygame_gui.elements.UIButton(relative_rect=pg.Rect((0, 0), (0, 0)), text='Step', manager=self.ui_manager)
+        self.step_button.tool_tip_text = "Shortcut: 'n'"
+        robots_list = ["Idle", "Random", "Left Wall Follower", "Right Wall Follower", "Flood Fill"]
```

Comment:
It's probably easier to use a `dict` (create it outside the class probably), then you don't need the `match` later.

Reply:
The `match` effectively implements a `dict` (return a fixed value based on a "key" from the list), and turning a `dict` to a list of keys is pretty simple (let's do `sorted(...)`).
The `match` is equivalent to `dict.get(key, idle_robot)`, let's keep the `get_robot...` function though, in case we need it.